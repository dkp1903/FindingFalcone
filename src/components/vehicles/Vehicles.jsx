import React, {Component} from 'react';
import ReactRadioButtonGroup from 'react-radio-button-group';

const Vehicles = (props) => {

  
    const {destination, vehicleClass, vehicles, vehSelection} = props.allProps;
    return (
      <div className={vehicleClass}>
        <ReactRadioButtonGroup
          name={vehicleClass}
          options={vehicles}
          value={vehSelection}
          onChange={ (event) => props.vehicleSelectFun(event, destination)}
        />
      </div>
    )
  }

  export default Vehicles
