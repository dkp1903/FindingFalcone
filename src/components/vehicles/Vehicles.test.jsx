import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import Vehicles from "./Vehicles.jsx";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

let propObject = {
  destination:"Destination1",
  vehicleClass:"vehicleRadioButton1",
  vehicles: [{value:"Space pot", label:"Space pot (2)"},{ value:"Rocket", label:"Rocket (1)"}],
  selectedVehicle: "Rocket"
}

it("renders successfully with props", () => {

  act(() => {
    render(<Vehicles
      vehicleSelectFun={() => {}}
      allProps={propObject} />, container);
  });
  expect(container).toMatchSnapshot();
});
