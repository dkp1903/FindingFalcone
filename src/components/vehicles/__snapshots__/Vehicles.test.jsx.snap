// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully with props 1`] = `
<div>
  <div
    class="vehicleRadioButton1"
  >
    <div>
      <div>
        <input
          id="react-radio-button-group-1-Space pot"
          name="vehicleRadioButton1"
          type="radio"
          value="Space pot"
        />
        <label
          for="react-radio-button-group-1-Space pot"
        >
          Space pot (2)
        </label>
      </div>
      <div>
        <input
          checked=""
          id="react-radio-button-group-1-Rocket"
          name="vehicleRadioButton1"
          type="radio"
          value="Rocket"
        />
        <label
          for="react-radio-button-group-1-Rocket"
        >
          Rocket (1)
        </label>
      </div>
    </div>
  </div>
</div>
`;
